### PR TITLE
Ethernet.begin() Low level test

### PR DESCRIPTION
**Pull request by Kavikick**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/spec/reports/
+vendor
+*.gem
+
+# rspec failure tracking
+.rspec_status
+
+# C++ stuff
+*.bin
+*.bin.dSYM
```
Comment (jgfoster) [inline .gitignore]: Add `.DS_Store` (unless you can explain what it is and why it should be committed). Also, Git likes to have an endline a the end of a file.

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+source 'https://rubygems.org'
+gem 'arduino_ci'
```
Comment (jgfoster) [inline Gemfile]: endline

File: test/test.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      int value = 0;
+      value = (value << 1) + state->digitalPin[SS];
+      value = (value << 1) + state->digitalPin[MOSI];
+      value = (value << 1) + state->digitalPin[MISO];
+      pinLog.push_back(value);
+      if (showData) {
+        std::cout.width(5);
+        std::cout << std::right << value << " : " << ((value >> 9) & 1) << "  "
+                  << ((value >> 8) & 1) << "  ";
+        // if (fourBitMode) {
```
Comment (jgfoster) [inline test/test.cpp]: Remove remnants from LiquidCrystal.

File: test/test.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  for (int i = 0; i < 52; i++) {
+    if (expected[i] != (int)state->spi.dataOut.at(i)) {
+      passed = false;
+      break;
+    }
+  }
+  assertTrue(passed);
+}
+
+unittest_main()
```
Comment (jgfoster) [inline test/test.cpp]: newline

**Comment by Kavikick** [comment]
Cleanup should be completed

**Pull request merged**